airgun/widgets.py

```diff
@@ -2238,3 +2238,54 @@ def items(self):
 
     def toggle(self, value):
         self.browser.click(self.ITEM.format(value))
+
+
+class InventoryBootstrapSwitch(Widget):
+    """Checkbox-like Switch control, representing On and Off state. But with
+    fancy UI and without any <form> elements.
+    There's also BootstrapSwitch widget in widgetastic_patternfly, but we don't
+    inherit from it as it uses completely different HTML structure than this one
+    (it has underlying <input>).
+    """
+
+    ON_TOGGLE = ".//span[contains(@class, 'bootstrap-switch-handle-on')]"
+    OFF_TOGGLE = ".//span[contains(@class, 'bootstrap-switch-handle-off')]"
+
+    def __init__(self, parent, class_name, **kwargs):
+        Widget.__init__(self, parent, logger=kwargs.pop('logger', None))
+        self.class_name = class_name
+
+    def __locator__(self):
+        return f"//div[@class='{self.class_name}']/div"
+
+    @property
+    def selected(self):
+        classes = self.browser.classes(self)
+        return 'bootstrap-switch-on' in classes
+
+    @property
+    def is_displayed(self):
```

Comment:
You shouldn't have to implement this. The base widget class already has it.

https://github.com/RedHatQE/widgetastic.core/blob/master/src/widgetastic/widget/base.py#L506